src/core/qgsvectorlayerupdater.h

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+                         qgsprocessingoutputs.h
+                         -------------------------
+    begin                : Feb 2018
+    copyright            : (C) 2018 by Arnaud Morvan
+    email                : arnaud dot morvan at camptocamp dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSVECTORLAYERUPDATER_H
+#define QGSVECTORLAYERUPDATER_H
+
+#include "qgis_core.h"
+#include "qgis.h"
+#include "qgsfeaturesink.h"
+#include "qgsvectorlayer.h"
+
+/**
+ * \class QgsVectorLayerUpdater
+ * \ingroup core
+ *
+ * Utility class to insert/update feature in an existing vector layer,
+ * using on source and destination fields mapping and primary key handling.
```

Comment:
Can you expand a bit more (and potentially even include some sample code in here)?